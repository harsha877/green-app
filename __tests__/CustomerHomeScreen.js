import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import CustomerHomeScreen from '../src/views/customerHomeScreen'

test("customer Home Screen Loading",()=>{
    const push = jest.fn();
   render(<CustomerHomeScreen navigation={{ push }} onSubmitHandler={()=>{}} user={{username:"sai",password:"123"}}/>)

})

test("customer HomeScreen snapshot",()=>{
    const push = jest.fn();
   let HomeComponenet=render(<CustomerHomeScreen navigation={{ push }} onSubmitHandler={()=>{}} user={{username:"sai",password:"123"}}/>).toJSON();
   expect(HomeComponenet).toMatchSnapshot();
})
describe("customer HomeScreen QueryTesting",()=>{
    const push = jest.fn();
    let HomeComponenet;
    beforeEach(()=>{
    HomeComponent=render(<CustomerHomeScreen navigation={{ push }} onSubmitHandler={()=>{}} user={{username:"sai",password:"123"}}/>);
    
    })

    test("Testing Title",()=>{
        const {getAllByText}=HomeComponent;
        expect(getAllByText("Customer Homepage").length).toBe(1);
     })
    
     test("Testing Text",()=>{
        const {getAllByText}=HomeComponent;
        expect(getAllByText("Create Quiz").length).toBe(1);
     })
     
     test("Testing Text",()=>{
        const {getAllByText}=HomeComponent;
        expect(getAllByText("View Quizzes").length).toBe(1);
     })
     
     test("Testing Text",()=>{
        const {getAllByText}=HomeComponent;
        expect(getAllByText("Log Out").length).toBe(1);
     })

     test("Testing Styles",()=>{
        const props=HomeComponent.toJSON();
        let CustomerTextStyles=props.children[0].props.style;
        expect(CustomerTextStyles).toStrictEqual({
            marginTop: 60,
            color: "#000000",
            fontSize: 20,
            fontWeight: "600",
        });
     })

     test("Testing Styles",()=>{
        const props=HomeComponent.toJSON();
        let CustomerButtonStyles=props.children[1].props.style;
        expect(CustomerButtonStyles).toStrictEqual({
            marginTop: 20,
            borderRadius: 50,
        });
     })
});


