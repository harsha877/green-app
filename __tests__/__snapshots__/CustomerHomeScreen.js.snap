// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer HomeScreen snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "margin": 60,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontSize": 20,
        "fontWeight": "600",
        "marginTop": 60,
      }
    }
  >
    Customer Homepage
  </Text>
  <View
    style={
      Object {
        "borderRadius": 50,
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#063f5c",
              },
            ]
          }
        >
          Create Quiz
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 50,
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#063f5c",
              },
            ]
          }
        >
          View Quizzes
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderRadius": 50,
        "marginTop": 100,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Log Out
        </Text>
      </View>
    </View>
  </View>
</View>
`;
